modules/widgets/src/theme-widget.tsx

```diff
@@ -135,23 +131,16 @@ export class ThemeWidget implements Widget<ThemeWidgetProps> {
       1,
       `Switching to ${this.themeMode === 'dark' ? this.props.darkModeLabel : this.props.lightModeLabel}`,
       themeStyle
-    )();
-
-    // Note: deck does not support updating the style property
-    // this.deck?.setProps({
-    //   style: themeStyle
-    // });
-    const elements = document.querySelectorAll('.deck-theme');
-    elements.forEach(root => {
-      const canvasStyle = (root as HTMLElement).style;
-      if (canvasStyle) {
-        for (const [variable, value] of Object.entries(themeStyle)) {
-          if (variable.startsWith('--')) {
-            canvasStyle.setProperty(variable, value);
-          }
-        }
+    );
+
+    const el = this.element;
+    if (el) {
+      const container = el.closest<HTMLDivElement>('.deck-widget-container');
```

Comment:
Nit:
- In terms of naming, I would elevate the theme as something separate from / "bigger than" widgets
- perhaps by calling the class `deck-theme` or `deck-theme-container`.

Reply:
Deck doesn't currently have styles outside of widgets. Do you have some in mind?

If we elevated it to the top-level of deck, there's a more more to consider: 

  - where the CSS class is applied within core. Currently, it's neatly scoped to the `WidgetManager` but if it's meant to apply to the rest of deck's DOM it should probably be added in `Deck` instead.
  - how it works in environments where widgets aren't supported (MapboxOverlay, GoogleMapsOverlay, etc..)

I'm just thinking if deck doesn't have styles outside of widgets, the next level up would be controlling the application's theme. Then you'd want the widget to behave like a `controlled` react input (`value`/`onChange` props).